### DISCUSSION TITLE
Unexpected Enum Value

### DISCUSSION THREAD

**Discussion by ghost** [sharksforarms/deku#361] [Q&A]
I have this enum:

```rust
#[derive(Debug, PartialEq, Clone, Copy, DekuRead, DekuWrite)]
#[deku(type = "u16")]
pub(crate) enum ResourceType {
   a = 1,
   b = 2,
   c = 3,
}
```

What if I receive a malformed but still valid input where ResourceType enum is 4 or any other value.
How can I ignore such cases or is there a way to handle myself by checking value and ignore if needed

**Comment by wcampbell0x2a** [comment]
The following will work:
```rust
#[derive(Debug, PartialEq, Clone, Copy, DekuRead, DekuWrite)]
#[deku(type = "u16")]
pub(crate) enum ResourceType {
    #[deku(id = "1")]
    A,
    #[deku(id = "2")]
    B,
    #[deku(id = "3")]
    C,
    #[deku(id_pat = "_")]
    Other(u16),
}
```

I'm sure we could figure out something in the future to support mostly "unit-only enum" with one `id_pat`.
